pkgs/tools/networking/openssh/common.nix

```diff
@@ -0,0 +1,117 @@
+{ pname
+, version
+, extraDesc ? ""
+, src
+, extraPatches ? []
+, extraNativeBuildInputs ? []
+}:
+
+{ lib, stdenv
+, pkgs
+, fetchurl
+, fetchpatch
+, zlib
+, openssl
+, libedit
+, pkg-config
+, pam
+, etcDir ? null
+, withKerberos ? true
+, kerberos
+, libfido2
+, withFIDO ? stdenv.hostPlatform.isUnix && !stdenv.hostPlatform.isMusl
+, linkOpenssl ? true
+}:
+
+with lib;
+stdenv.mkDerivation rec {
+  inherit pname version src;
+
+  patches =
+    [
+      ./locale_archive.patch
+
+      # See discussion in https://github.com/NixOS/nixpkgs/pull/16966
+      ./dont_create_privsep_path.patch
+    ]
+    ++ extraPatches;
+
+  postPatch =
+    # On Hydra this makes installation fail (sometimes?),
+    # and nix store doesn't allow such fancy permission bits anyway.
+    ''
+      substituteInPlace Makefile.in --replace '$(INSTALL) -m 4711' '$(INSTALL) -m 0711'
+    '';
+
+  nativeBuildInputs = [ pkg-config ]
+    ++ optional withKerberos pkgs.kerberos.dev
```

Comment:
```suggestion
    ++ optional withKerberos kerberos.dev
```

Reply:
Could you also check if the .dev is not needed? I think build inputs use the .dev output by default.

Reply:
Oof my system has garbage-collected the build and its dependencies :/ I might get around to do that in the next days, though

Reply:
I can probably build this in under an hour.